### PR TITLE
Add DELETE to the _localstack/ses/ endpoint

### LINKED ISSUE #7491: feature request: implement DELETE /_localstack/ses endpoint
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

Provide an URL where I can delete stored SES emails.
My motivation is to write integration tests much easier, if I could delete the stored emails from the localstack before each test case. 

I use the docker image to run localstack. Mount the `/var/lib/localstack` folder to the host and delete the files from the `tmp/state/ses` is cumbersome because of the file permission issue of docker.

I think the simplest way to achieve the empty SES emails state is to implement a new endpoint in a same way as to the GET /_localstack/ses has been implemented:

**path:** /_localstack/ses
**method:** DELETE

### 🧑‍💻 Implementation

I am not a python developer, but I think one solution is to extend the [SesServiceApiResource](https://github.com/localstack/localstack/blob/590de96b3dd65e349709300b668e70f1f9664e99/localstack/services/ses/provider.py#L121) with an `on_delete` method.

Open question for me is either delete the `/var/lib/localstack/tmp/state/ses` folder or just delete the files of the `ses` folder.
The folder does not exist when the container starts. It is created when the first email received and saved by the service.
I think both solutions have the same performance impact, so the real question is what is preferred by the maintainers.

As I mentioned I am not familiar with python ecosystem but I am happy to help implementing the feature if you accept the request.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by justinwyer**
Resolves #7491 

Added the following endpoints:
- `DELETE _localstack/ses` requested by the issue
- `DELETE _localstack/ses?id=<message_id>` [requested in a comment](https://github.com/localstack/localstack/issues/7491#issuecomment-1414126917) 
- `GET       _localstack/ses?id=<message_id>` added to match the DELETE by id

### CODE REVIEW

**Comment by justinwyer** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by justinwyer** [comment]
As part of the testing I was using the `_aws/ses` endpoint but had to switch to `_localstack/ses` as the `message_id` parameter in the route `_localstack/ses/<message_id>` parameter was not working with the `_aws` prefix. If you can shed some light as to why this might be I would be happy to fix that.

**Comment by justinwyer** [comment]
I am not sure what Circle CI is upset about exactly, perhaps someone can rerun the workflow?

File: localstack/services/ses/provider.py [hunk 1]
```diff
@@ -128,7 +128,10 @@ class SesServiceApiResource:
     - GET param `email`: filter for `source` field in SES message
     """
 
-    def on_get(self, request):
+    def on_get(self, request, message_id=None):
+        if message_id is not None:
+            return EMAILS[message_id]
```
Comment (viren-nadkarni) [inline localstack/services/ses/provider.py:133]: Let's keep the response schema the same as before, whether or not the `message_id` filter is specified.

```
{
  "messages": [
    {<the message>}
  ]
}
```

File: localstack/services/ses/provider.py [hunk 2]
```diff
@@ -128,7 +128,10 @@ class SesServiceApiResource:
     - GET param `email`: filter for `source` field in SES message
     """
 
-    def on_get(self, request):
+    def on_get(self, request, message_id=None):
```
Comment (viren-nadkarni) [inline localstack/services/ses/provider.py:131]: I would make `message_id` a query parameter which is more backward compatible.

Could we also update the docstring? Also the behaviour when both `message_id` and `email` are provided.

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]

File: tests/integration/test_ses.py [hunk 3]
```diff
@@ -181,9 +181,12 @@ def _read_message_from_filesystem(message_id: str) -> dict:
         assert "Body" not in contents2
 
         # Ensure all sent messages can be retrieved using the API endpoint
-        emails_url = config.get_edge_url() + EMAILS_ENDPOINT
-        api_contents = requests.get(emails_url).json()
-        api_contents = {msg["Id"]: msg for msg in api_contents["messages"]}
+        emails_url = config.get_edge_url() + "/_localstack/ses/"
```
Comment (viren-nadkarni) [inline tests/integration/test_ses.py:184]:
```suggestion
        emails_url = config.get_edge_url() + EMAILS_ENDPOINT
```

File: tests/integration/test_ses.py [hunk 4]
```diff
@@ -181,9 +181,12 @@ def _read_message_from_filesystem(message_id: str) -> dict:
         assert "Body" not in contents2
 
         # Ensure all sent messages can be retrieved using the API endpoint
-        emails_url = config.get_edge_url() + EMAILS_ENDPOINT
```
Comment (viren-nadkarni) [inline tests/integration/test_ses.py:184]: We need to test the 'get all messages' functionality as before.

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]
Thanks for addressing my comments. Just a few minor points and we can merge this :+1:

File: tests/integration/test_ses.py (hunk 4 above)
Comment (justinwyer) [reply tests/integration/test_ses.py:184]: This is still tested in the `test_send_templated_email_can_retrospect` above.

**Review by viren-nadkarni — APPROVED** [review]

**Pull request merged**